src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -459,6 +491,34 @@ internal static unsafe int CompareOrdinalIgnoreCase(string strA, int indexA, int
                     length--;
                 }
 
+                if (CultureData.InvariantMode)
```

Comment:
I didn't do that for the perf reason. this will ass extra check with every character.